checker-rs/src/main.rs

```diff
@@ -0,0 +1,327 @@
+use enochecker::{
+    async_trait,
+    tokio::{spawn, try_join},
+};
+use enochecker::{run_checker, Checker, CheckerError, CheckerRequest, CheckerResult};
+
+use fake::{
+    faker::internet::en::{FreeEmail, Password, SafeEmail, Username},
+    Fake,
+};
+
+use rand::random;
+use serde::{Deserialize, Serialize};
+
+use mongodb::{
+    bson::doc,
+    options::{ClientOptions, StreamAddress},
+    Client,
+};
+
+use tracing::{error, info, warn, Instrument};
+
+mod notebook_client;
+use notebook_client::NotebookClient;
+
+struct NotebookChecker {
+    db: Client,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct NotebookUser {
+    username: String,
+    password: String,
+    note_id: Option<String>,
+    note: Option<String>,
+    task_chain_id: String,
+}
+
+impl NotebookUser {
+    fn gen_random(id: String) -> Self {
+        let username: String = if rand::random() {
+            if rand::random() {
+                FreeEmail().fake()
+            } else {
+                SafeEmail().fake()
+            }
+        } else {
+            Username().fake()
+        };
+
+        Self {
+            username,
+            password: Password(12..24).fake::<String>(),
+            note_id: None,
```

Comment:
yes, since it makes it less failure prune since you can enforce the right state, but the gtfo everywhere if you're in the wrong state makes it also a bit annoying to work with